rules/TypeDeclaration/Rector/ClassMethod/ReturnTypeFromMockObjectRector.php

```diff
@@ -101,12 +103,14 @@ public function refactorWithScope(Node $node, Scope $scope): ?Node
             return null;
         }
 
-        $soleReturn = $returns[0];
-        if (! $soleReturn->expr instanceof Expr) {
+        if (! $this->returnAnalyzer->hasOnlyReturnWithExpr($node, $returns)) {
             return null;
         }
 
-        $returnType = $this->getType($soleReturn->expr);
```

Comment:
I wonder whethere there are real use-cases for `getType()` anymore, or whether this method should be deprecated in favor of `getNativeType`...?

Reply:
`getType()` is stil usefull for:

- get type from php native function/methods, yes, it got mixed on getNativeType()
-  downgrade process
- verify transformation that too much for only native check, eg,

```php
/** @var string $var */
trim($var)
 ```

that cast is not needed on that case.

Reply:
Ok, said differently: its only useful in very few edge cases but the name misleads even rector-src core developers to use it wrongly in lots of situations

Reply:
Given rare context, this is used on purpose here 👍